### PR TITLE
Refactor the supporting library

### PR DESCRIPTION
**Pull request by koher**
To make it easier to create `FunctionCallExpression`s from the Translate module.

### CODE REVIEW

File: Sources/SupportingLibrary.swift [hunk 1]
```diff
@@ -1,92 +1,104 @@
...
+                name =  "qq"
+                declaration = "function qq(expression, value) {\n  const is_null = (expression == null || typeof expression == 'undefined');\n  return is_null ? value : expression;\n}"
             }
         }
+        
+        func callExpression(expression: Expression, arguments: [Expression], namepace: String = SupportingLibrary.defaultNamespace) -> FunctionCallExpression {
+            return FunctionCallExpression(expression: ExplicitMemberExpression(expression: expression, member: name), arguments: arguments.map { (nil, $0) }, trailingClosure: nil)
+        }
+        
+        enum Id {
```
Comment (masarusanjp) [inline Sources/SupportingLibrary.swift:59]: Good name 👍

File: Sources/SupportingLibrary.swift [hunk 2]
```diff
@@ -1,92 +1,104 @@
...
-        case optionalChaining
-        case range
-        case closedRange
-        case typeChecking
-        case nilCoalescing
-
-        var function: Function {
-            switch self {
+        
+        private static let idToFunction: [Id: Function] = {
```
Comment (masarusanjp) [inline Sources/SupportingLibrary.swift:8]: [nits] Do you use this variable? It seems not using.

**Review by masarusanjp — COMMENTED** [review]
Looks good 👍 
I commented a followings 

- `idToFunction` seems not using.

File: Sources/SupportingLibrary.swift (hunk 2 above)
Comment (koher) [reply Sources/SupportingLibrary.swift:8]: Thanks. I fixed the initializer to use `idToFunction`.

**Comment by koher** [comment]
👍

**Comment by masarusanjp** [comment]
👍

**Pull request merged**